### PR TITLE
[FAI-254] Add counterfactual explainer using OptaPlanner

### PR DESCRIPTION
**Pull request by ruivieira**
Same codebase as [PR 479](https://github.com/kiegroup/kogito-apps/pull/479) to `kogito-apps`.
Using the `sandbox` for the meantime to continue development.

### CODE REVIEW

**Comment by ruivieira** [comment]
@danielezonca, @tteofili ptal, thank you!

File: explainability/core/counterfactuals/pom.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            <dependency>
+                <groupId>org.optaplanner</groupId>
+                <artifactId>optaplanner-core</artifactId>
+                <version>${version.org.optaplanner}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.kie.kogito</groupId>
+                <artifactId>explainability-core</artifactId>
+                <version>1.0.0-SNAPSHOT</version>
+            </dependency>
```
Comment (danielezonca) [inline explainability/core/counterfactuals/pom.xml:28]: I don't think we publish -SNAPSHOT artifacts to public Maven repositories so you might need to add this repository
https://github.com/kiegroup/kogito-runtimes/blob/master/pom.xml#L67-L84

This should be ok also for OptaPlanner dependency

Comment (ruivieira) [reply explainability/core/counterfactuals/pom.xml]: Updated POM

**Review by danielezonca — APPROVED** [review]
LGTM :+1:

**Review by danielezonca — APPROVED** [review]

**Pull request merged**